core.cc

```diff
@@ -90,7 +90,7 @@ NAN_METHOD(SysLog) {
 
     if (message) {
         Nan::AsyncQueueWorker(new Worker(callback, priority, message));
-    } else if(!callback->IsEmpty()) {
+    } else if(callback->GetFunction()->IsFunction()) {
```

Comment:
Don't do `info[2].As<Function>()` on line 83 but check that `info[2]->IsFunction()` is true, then you know whether you have a function or not.

Aside: there is a memory leak here, `callback` is not freed.